### DISCUSSION TITLE
Miscelanious feature requests/suggestions

### DISCUSSION THREAD

**Discussion by LadyHylia** [Valkirie/HandheldCompanion#1065] [Ideas]
I've been using handheld companion for about a week now as a replacement for armory crate on my ROG Ally(Gen 1 non-extreme), and while it wins over ac by default for letting me rebind the oem buttons, I couldn't help but come up with some ways to improve it further. I have no idea how feasible any of these ideas are, but I thought I would share them in case they end up being of use. Most of these are ideas for quick tools but I have some ideas for the rest of the program too. 

-Custom quick tools buttons

-Quick tool button to cycle through power settings. Currently, as far as I can tell, there is no way to change power plans with a controller. When going into profiles > power plans in the quick tools menu, pressing "A" on a power plan edits it, and doesn't enable it.

-Multiple triggers per hotkey (I.E. open quick tools with an oem button and a keyboard shortcut)

-Quick tool button to kill active task

-Change side of screen quick tools appears on (this is shown in the github readme but I can't find the option in the program so I'm mentioning it here)

-Map gyro sensors directly to virtual DS4 gyro for greater compatibility with Steam Input

-Quick tool button for flashback recording and taking a screenshot

-Deeper options for hotkey customization for parity with controller mapping options (I.E. press button twice to open quick menu, map hotkey to virtual gamepad buttons)

-Remember specific controllers and connect/apply custom profiles to them automatically

-Disconnect button for gamepads. If it exists, make it more obvious for us hard of noticing

-Silent quick tools open/close (it currently makes the windows "click" sound for me, not sure if this is a bug)

-Custom quick tools background

Thats it for now, but I'll update this post if I think of more things. Thank you devs for this amazing program, I can't wait to see how it evolves as the handheld pc gaming scene continues to grow! I don't do this often, but I will be donating some to support the development once I get my next paycheck. This program is important for the future of the scene and I'd like to support its development any way I can. If I knew any "coding" beyond scripting in Skyrim's papyrus language, I would throw my hat in to help add these suggestions, but right now all I can do is suggest and monetarily support.

EDIT 6/29/2024: I have a few more suggestions I've been gathering, but I noticed in the readme that you prefer getting them in the discord. I feel like I would be spamming if I dumped them all at once, so I'll try to space them out...

**Comment by CasperH2O** [comment]
Hey @LadyHylia thanks for dropping in, see below in line my replies.

-Custom quick tools buttons

You can have 10 custom quick tool buttons, see bottom of the Hotkeys page, use the pin to pin them to QT on the right side.

-Quick tool button to cycle through power settings. Currently, as far as I can tell, there is no way to change power plans with a controller. When going into profiles > power plans in the quick tools menu, pressing "A" on a power plan edits it, and doesn't enable it.

You need to press (Y) to apply them with gamepad navigation. In the main UI there's a prompt but that's missing from QT.

-Multiple triggers per hotkey (I.E. open quick tools with an oem button and a keyboard shortcut)

Interesting.

-Quick tool button to kill active task

It's there, you may need to pin it though, it's the X item.

-Change side of screen quick tools appears on (this is shown in the github readme but I can't find the option in the program so I'm mentioning it here)

-Map gyro sensors directly to virtual DS4 gyro for greater compatibility with Steam Input

That's already done, using the Virtual DS4 will get you your selected IMU (internal or joycon or external) so you can use it in games or Steam

-Quick tool button for flashback recording and taking a screenshot

-Deeper options for hotkey customization for parity with controller mapping options (I.E. press button twice to open quick menu, map hotkey to virtual gamepad buttons)

-Remember specific controllers and connect/apply custom profiles to them automatically

-Disconnect button for gamepads. If it exists, make it more obvious for us hard of noticing

It's in the first page in the main GUI next to the controllers.

-Silent quick tools open/close (it currently makes the windows "click" sound for me, not sure if this is a bug)

Owh, that might be a bug. Unless you enabled the Setting for sound?

-Custom quick tools background

Ha, back to the Winamp days with custom skins! HC has several themes, perhaps a different one is more to your liking (dark, light, myca etc)

**Reply by LadyHylia** [reply]
Oh jeez, I didn't realize the hotkeys section let you pin them to the quick tools, so that's around half my suggestions. Thanks for clearing that up, I feel like I wouldn't have noticed that for a long time. Maybe rename that section of the main window to quick tools? Or put a quick explanation of the buttons on screen somewhere, just to prompt the user to fully consider whats in front of them. I know I noticed the pin button, but I never even thought about what it might do. 

"-Quick tool button to cycle through power settings. Currently, as far as I can tell, there is no way to change power plans with a controller. When going into profiles > power plans in the quick tools menu, pressing "A" on a power plan edits it, and doesn't enable it.

You need to press (Y) to apply them with gamepad navigation. In the main UI there's a prompt but that's missing from QT."

Oh that makes sense. It wouldn't have occurred to me that the button prompts would be the same across the main window and quick tools.

"-Map gyro sensors directly to virtual DS4 gyro for greater compatibility with Steam Input

That's already done, using the Virtual DS4 will get you your selected IMU (internal or joycon or external) so you can use it in games or Steam"

Oh ok, I wasn't aware of this. I think it might not be working for me, as I couldn't get steam input to recognize the virtual ds4 gyro at all in my testing, and I didn't see anything in the menu that concretely said thats what it was trying to do. I found options to map it to sticks and whatnot, but nothing that said like "map to raw ds4 gyro" or something.

"-Disconnect button for gamepads. If it exists, make it more obvious for us hard of noticing

It's in the first page in the main GUI next to the controllers."

I'm not sure if this is just responding to that last point, but I don't see any option to do this. Is it the unhide button? If so it wasn't clear to me that this was what it did. Do I re-connect the attached controller to detach the external one? I really just want to turn my external controller off without crashing HC and performing a restart.

"-Silent quick tools open/close (it currently makes the windows "click" sound for me, not sure if this is a bug)

Owh, that might be a bug. Unless you enabled the Setting for sound?"

Yeah sound is disabled in settings.

"-Custom quick tools background

Ha, back to the Winamp days with custom skins! HC has several themes, perhaps a different one is more to your liking (dark, light, myca etc)"

The winamp days never ended if its still alive in your heart! Every program needs an option to open file explorer and choose a .png to display in the background!

Also while looking through your answers I found a bug.  I'll make a post about this in the proper section but I figure I'll mention it here. I can't change the power plan through HC at all. Whenever I tell it to switch to a new one, it switches back to default immediately. I also get two windows toast notifications, so it seems like something is actively changing it back and its not just the ui bugging out.
